From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: RENaa_FD <92320175+lRENyaaa@users.noreply.github.com>
Date: Tue, 26 Aug 2025 04:00:56 +0800
Subject: [PATCH] Backport Player#isConnected to PandaSpigot

PacketEvents sometimes use this method, but it seems they did not account for compatibility with older minecraft versions. So we backport the Player#isConnected method intended for personal use.

diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index 49477a0f4d237db75d869ed91dc5088f4b4385fe..3396a48e6801f4b085c2ee415477daba83bf84ea 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -950,6 +950,11 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
         return null;
     }
 
+    @Override
+    public boolean isConnected() {
+        return getHandle().playerConnection != null && !getHandle().playerConnection.isDisconnected();
+    }
+
     @Override
     public void setBedSpawnLocation(Location location) {
         setBedSpawnLocation(location, false);
